src/plugins/resolver/filename.c

```diff
@@ -406,11 +422,26 @@ static int elektraResolveSpec (resolverHandle * p, Key * warningsKey ELEKTRA_UNU
 	}
 	else
 	{
-		strcpy (p->filename, KDB_DB_SPEC);
-		strcat (p->filename, "/");
-		strcat (p->filename, p->path);
+		if (KDB_DB_SPEC[0] == '~')
```

Comment:
Did you already consider discussion at #691? ~ would not be on the first position.